src/utils/common.ts

```diff
@@ -1,26 +1,34 @@
+import { isRouteErrorResponse } from '@remix-run/react';
+import { isEcomSDKError } from '~/api/types';
+
 export function getUrlOriginWithPath(url: string) {
     const { origin, pathname } = new URL(url);
     return new URL(pathname, origin).toString();
 }
 
-export function toError(value: unknown): Error {
-    if (value instanceof Error) {
-        return value;
+/**
+ * Try to find some error message in unknown value.
+ *
+ * Handles cases when value is `ErrorResponse`, `EcomSDKError`, `String` or `Object`.
+ *
+ * Falls back to 'Unknown error' message.
+ */
+export function getErrorMessage(value: unknown): string {
+    if (isRouteErrorResponse(value)) {
+        return String(value.data);
     }
 
-    if (typeof value === 'undefined' || value === null) {
-        return new Error();
+    if (value instanceof Error || isEcomSDKError(value)) {
+        return value.message;
     }
 
-    if (typeof value === 'object') {
-        if ('message' in value) {
-            throw new Error(String(value.message));
-        }
+    if (typeof value === 'object' && value !== null) {
+        return JSON.stringify(value);
+    }
```

Comment:
What's this for?

Reply:
this is needed to handle such cases:
<img width="710" alt="image" src="https://github.com/user-attachments/assets/146200bf-ee66-4d70-9717-4847f6699b6f">

instead of showing [Object object] (in case String(value))
![image](https://github.com/user-attachments/assets/c2f694d6-3e15-4607-b069-7c2aa86875e8)

Reply:
First, if someone is throwing a value that contains garbage they shouldn't expect to get a nice error message.

Second, the `JSON.stringify()` here can throw.

So I don't think this is needed at all.

Reply:
Regarding the first point - maybe somebody who is throwing garbage shouldn't expect a nice error message, but user should get at least something different from 'Unknown error' if it's possible, to provide to customer support.

Reply:
so I'd rather wrap this call in `try...catch` and return JSON string when possible

Reply:
@yurii-ve ok, makes sense